libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
Agree it would be very rare, just thought it was worth acknowledging that this scheme isn't foolproof.

No need to instrument anything, would be a pain to plumb the log through.  A comment somewhere is good enough.